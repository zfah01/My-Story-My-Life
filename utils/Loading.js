import React from 'react';
import { ImageBackground } from 'react-native';
import Spinner from 'react-native-loading-spinner-overlay';

//Application Logo is displayed when application is loading 
export default function Loading(props) {
    const loading = props.loading;
    return (
        <ImageBackground source={require('../assets/logo.png')} style={{ width: '100%', height: '70%', opacity: 50, marginTop: 140}} >
            <Spinner
            visible={loading}
            textContent={'Loading...'}
            textStyle={{color: '#FFFFFF'}} />
        </ImageBackground>
    );
}
