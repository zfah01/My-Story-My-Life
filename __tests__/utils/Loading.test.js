import 'react-native';
import React from 'react';
import Loading from '../../utils/Loading';

import renderer from 'react-test-renderer';

it('Test to check that Loading is rendered', () => {
  const tree = renderer.create(<Loading />).toJSON();
  expect(tree).toMatchSnapshot();
});