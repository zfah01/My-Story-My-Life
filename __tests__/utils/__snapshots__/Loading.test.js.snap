// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks to see if the Loading Page renders 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "70%",
      "marginTop": 140,
      "opacity": 50,
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "70%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    statusBarTranslucent={true}
    supportedOrientations={
      Array [
        "landscape",
        "portrait",
      ]
    }
    transparent={true}
    visible={false}
  />
</View>
`;

exports[`Test to check that Loading is rendered 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "70%",
      "marginTop": 140,
      "opacity": 50,
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "70%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    statusBarTranslucent={true}
    supportedOrientations={
      Array [
        "landscape",
        "portrait",
      ]
    }
    transparent={true}
    visible={false}
  />
</View>
`;
